hi-ml-cpath/src/health_cpath/datamodules/base_module.py

```diff
@@ -273,47 +272,29 @@ class SlidesDataModule(HistoDataModule[SlidesDataset]):
 
     def __init__(
         self,
-        level: int = 1,
-        backend: str = 'cuCIM',
-        wsi_reader_args: Dict[str, Any] = {},
+        loading_params: LoadingParams = LoadingParams(),
         tiling_params: TilingParams = TilingParams(),
         **kwargs: Any,
     ) -> None:
         """
-        :param level: the whole slide image level at which the image is extracted, defaults to 1
-        this param is passed to the LoadImaged monai transform that loads a WSI with cucim backend by default
-        :param backend: the WSI reader backend, defaults to "cuCIM".
-        :param wsi_reader_args: additional arguments to pass to the WSIReader, defaults to {}. Multi processing is
-        enabled since monai 1.0.0 by specifying num_workers > 0 with CuCIM backend only.
         :param tiling_params: the tiling on the fly parameters, defaults to TileOnTheFlyParams()
+        :param loading_params: the loading parameters, defaults to LoadingParams()
+        :param kwargs: additional parameters to pass to the parent class HistoDataModule
         """
         super().__init__(**kwargs)
         self.tiling_params = tiling_params
-        # WSIReader params
-        self.level = level
-        self.backend = backend
-        self.wsi_reader_args = wsi_reader_args
+        self.loading_params = loading_params
 
     def _load_dataset(self, slides_dataset: SlidesDataset, stage: ModelKey) -> Dataset:
         base_transform = Compose(
             [
-                LoadImaged(
-                    keys=SlideKey.IMAGE,
-                    reader=WSIReader,
-                    dtype=np.uint8,
-                    image_only=True,
-                    level=self.level,
-                    backend=self.backend,
-                    **self.wsi_reader_args,
-                ),
+                self.loading_params.get_load_roid_transform(),
                 self.tiling_params.get_tiling_transform(bag_size=self.bag_sizes[stage], stage=stage),
-                # GridPatchd returns stacked tiles (bag_size, C, H, W), however we need to split them into separate
-                # tiles to be able to apply augmentations on each tile independently
-                SplitDimd(keys=SlideKey.IMAGE, dim=0, keepdim=False, list_output=True),
+                self.tiling_params.get_split_transform(),
```

Comment:
moved to tiling_params to be able to test it. Given that args are hardcoded, we test that they are indeed what we expect, easier to catch if something goes wrong